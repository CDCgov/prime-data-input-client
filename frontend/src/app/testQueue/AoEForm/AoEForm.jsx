import React from "react";
import { displayFullName } from "../../utils";
import { useState } from "react";
import {
  getSymptomList,
  getTestTypes,
  getPregnancyResponses,
} from "./constants";
import { COVID_RESULTS, TEST_RESULT_DESCRIPTIONS } from "../../constants";
import Checkboxes from "../../commonComponents/Checkboxes";
import RadioGroup from "../../commonComponents/RadioGroup";
import Dropdown from "../../commonComponents/Dropdown";
import Button from "../../commonComponents/Button";
import TextInput from "../../commonComponents/TextInput";
import FormGroup from "../../commonComponents/FormGroup";
import Optional from "../../commonComponents/Optional";
import { testResultQuery } from "../../testResults/TestResultsList";
import { useQuery } from "@apollo/client";
import moment from "moment";

// Get the value associate with a button label
// TODO: move to utility?
const findValueForLabel = (label, list) =>
  (list.filter((item) => item.label === label)[0] || {}).value;

// building blocks that *probably* don't want to be exportable components
const SymptomInputs = ({
  symptomListConfig,
  noSymptoms,
  setNoSymptoms,
  currentSymptoms,
  setSymptoms,
  onsetDate,
  setOnsetDate,
  symptomError,
  symptomOnsetError,
}) => {
  const symptomChange = (e) => {
    setSymptoms({ ...currentSymptoms, [e.target.value]: e.target.checked });
  };

  return (
    <>
      <div className={symptomError ? "usa-form-group--error" : "usa-fieldset"}>
        <Checkboxes
          name="symptom_list"
          legend="Are you experiencing any of the following symptoms?"
          onChange={(e) => setNoSymptoms(e.target.checked)}
          boxes={[{ value: "no", label: "No Symptoms", checked: noSymptoms }]}
          required
          errorMessage={symptomError}
          validationStatus={symptomError ? "error" : null}
        />
        {!noSymptoms && (
          <>
            <div className="border-top-1px border-base-lighter margin-top-2 margin-bottom-05"></div>
            <Checkboxes
              legend="What are your symptoms?"
              legendSrOnly
              name="symptom_list"
              onChange={symptomChange}
              boxes={symptomListConfig.map(({ label, value }) => ({
                label,
                value,
                checked: currentSymptoms[value],
              }))}
            />
          </>
          )}
      </div>
      {!noSymptoms && (
        <TextInput
          type="date"
          label="Date of symptom onset"
          name="symptom_onset"
          value={onsetDate}
          onChange={(e) => setOnsetDate(e.target.value)}
          min="2020-02-01"
          max={new Date().toISOString().split("T")[0]}
          required={Object.keys(currentSymptoms).some(
            (key) => currentSymptoms[key]
          )}
          errorMessage={symptomOnsetError}
          validationStatus={symptomOnsetError ? "error" : null}
        />
      )}
    </>
  );
};

const PriorTestInputs = ({
  testTypeConfig,
  isFirstTest,
  setIsFirstTest,
  priorTestDate,
  setPriorTestDate,
  priorTestResult,
  setPriorTestResult,
  priorTestType,
  setPriorTestType,
  mostRecentTest,
}) => {
  const recentDate = (mostRecentTest?.dateTested || "").split("T")[0];
  const filledPriorTest =
    priorTestDate &&
    recentDate === priorTestDate &&
    mostRecentTest.result === priorTestResult;
  const [mostRecentTestAnswer, setMostRecentTestAnswer] = useState(
    !priorTestDate || isFirstTest === undefined
      ? undefined
      : filledPriorTest
      ? "yes"
      : "no"
  );
  const previousTestEntry = (
    <>
      <TextInput
        type="date"
        label="Date of Most Recent Test"
        name="prior_test_date"
        value={priorTestDate}
        onChange={(e) => setPriorTestDate(e.target.value)}
        max={new Date().toISOString().split("T")[0]}
        min="2020-02-01"
      />
      <Dropdown
        options={testTypeConfig}
        label="Type of Prior Test"
        name="prior_test_type"
        selectedValue={priorTestType}
        onChange={(e) => setPriorTestType(e.target.value)}
        defaultSelect
      />
      <Dropdown
        options={[
          {
            value: COVID_RESULTS.POSITIVE,
            label: TEST_RESULT_DESCRIPTIONS.POSITIVE,
          },
          {
            value: COVID_RESULTS.NEGATIVE,
            label: TEST_RESULT_DESCRIPTIONS.NEGATIVE,
          },
          {
            value: COVID_RESULTS.INCONCLUSIVE,
            label: TEST_RESULT_DESCRIPTIONS.UNDETERMINED,
          },
        ]}
        label="Result of Prior Test"
        name="prior_test_result"
        selectedValue={priorTestResult}
        defaultSelect
        onChange={(e) => setPriorTestResult(e.target.value)}
      />
    </>
  );

  // If mostRecentTest matches priorTest, offer to fill it
  if (mostRecentTest) {
    // Suggest a prior test
    // TODO: ARIA/508 compliance
    const legendIsh = (
      <>
        <div className="margin-top-2">
          <b>Was this your most recent COVID-19 test?</b>
          <Optional/>
        </div>
        <p className="prime-previous-test-display margin-top-2 margin-bottom-0 line-height-sans-5">
          <b>Date: </b>
          {moment(mostRecentTest.dateTested).format("LLLL")}<br />
          <b>Type: </b>
          Antigen<br />
          <b>Result: </b>
          {mostRecentTest.result}
        </p>
      </>
    );
    return (
      <>
        {legendIsh}
        <RadioGroup
          buttons={[
            { label: "Yes", value: "yes" },
            { label: "No", value: "no" },
          ]}
          selectedRadio={mostRecentTestAnswer}
          onChange={(e) => {
            setIsFirstTest(false);
            setMostRecentTestAnswer(e.target.value);
            if (e.target.value === "yes") {
              // Fill in last test info using this data
              // TODO: update when test history has test type
              setPriorTestType("2");
              setPriorTestDate((mostRecentTest.dateTested || "").split("T")[0]);
              setPriorTestResult(mostRecentTest?.result);
            } else {
              setPriorTestType(null);
              setPriorTestDate(null);
              setPriorTestResult(null);
            }
          }}
          legend="Was this your most recent COVID-19 test?"
          legendSrOnly
          name="most_recent_flag"
          variant="horizontal"
        />
        {mostRecentTestAnswer === "no" && previousTestEntry}
      </>
    );
  }

  return (
    <>
      <div className="usa-legend">
        Is this your first COVID-19 test?
        <Optional />
      </div>
      <div className="prime-previous-test-display margin-top-2">
        SimpleReport did not find any previous test data
      </div>

      <RadioGroup
        buttons={[
          { label: "Yes", value: "yes" },
          { label: "No", value: "no" },
        ]}
        selectedRadio={
          isFirstTest === true ? "yes" : isFirstTest === false ? "no" : ""
        }
        onChange={(e) => {
          setIsFirstTest(e.target.value === "yes");
        }}
        legend="Is this your first covid test?"
        legendSrOnly
        name="prior_test_flag"
        variant="horizontal"
      />
      {isFirstTest === false && previousTestEntry}
    </>
  );
};

const AoEForm = ({
  saveButtonText,
  onClose = {},
  patient,
  facilityId,
  loadState = {},
  saveCallback = {},
  isModal,
}) => {
  // this seems like it will do a bunch of wasted work on re-renders and non-renders,
  // but it's all small-ball stuff for now
  const testConfig = getTestTypes();
  const symptomConfig = getSymptomList();
  const initialSymptoms = {};
  if (loadState.symptoms) {
    const loadedSymptoms = JSON.parse(loadState.symptoms);

    symptomConfig.forEach((opt) => {
      const val = opt.value;
      if (typeof loadedSymptoms[val] === "string") {
        initialSymptoms[val] = loadedSymptoms[val] === "true";
      } else {
        initialSymptoms[val] = loadedSymptoms[val];
      }
    });
  } else {
    symptomConfig.forEach((opt) => {
      initialSymptoms[opt.value] = false;
    });
  }

  const [noSymptoms, setNoSymptoms] = useState(loadState.noSymptoms || false);
  const [currentSymptoms, setSymptoms] = useState(initialSymptoms);
  const [onsetDate, setOnsetDate] = useState(loadState.symptomOnset);
  const [isFirstTest, setIsFirstTest] = useState(loadState.firstTest);
  const [priorTestDate, setPriorTestDate] = useState(loadState.priorTestDate);
  const [priorTestType, setPriorTestType] = useState(loadState.priorTestType);
  const [priorTestResult, setPriorTestResult] = useState(
    loadState.priorTestResult
  );
  const [pregnancyResponse, setPregnancyResponse] = useState(
    loadState.pregnancy
  );

  // form validation
  const [symptomError, setSymptomError] = useState(null);
  const [symptomOnsetError, setSymptomOnsetError] = useState(null);


  const isValidForm = () => {
    const hasSymptoms = Object.keys(currentSymptoms).some((key) => currentSymptoms[key]);
    if (!noSymptoms && !hasSymptoms) {
      setSymptomError('Select your symptoms')
      setSymptomOnsetError(null)
      return false;
    }

    if (noSymptoms) {
      setSymptomError(null)
      setSymptomOnsetError(null)
      return true;
    }

    if (hasSymptoms && !onsetDate) {
      setSymptomError(null);
      setSymptomOnsetError("Enter the date of symptom onset");
      return false;
    }

    if (hasSymptoms && onsetDate) {
      setSymptomError(null);
      setSymptomOnsetError(null);
      return true;
    }
  };

  // TODO: only get most recent test from the backend
  const { data, loading, error } = useQuery(testResultQuery, {
    fetchPolicy: "no-cache",
    variables: { facilityId },
  });
  if (loading) return null;
  if (error) throw error;
  const mostRecentFirst = (a, b) => {
    const ma = moment(a.dateTested);
    const mb = moment(b.dateTested);
    if (ma && ma.isAfter(mb)) return -1;
    if (ma && ma.isBefore(mb)) return 1;
    return 0;
  };
  const mostRecentTest = data.testResults
    .filter((r) => r.patient.internalId === patient.internalId)
    .sort(mostRecentFirst)[0];

  // Auto-answer pregnancy question for males
  const pregnancyResponses = getPregnancyResponses();
  if (patient.gender === "male" && !pregnancyResponse) {
    setPregnancyResponse(findValueForLabel("No", pregnancyResponses));
  }

  const saveAnswers = (e) => {
    if (isValidForm()) {
      const saveSymptoms = { ...currentSymptoms };
      if (noSymptoms) {
        // set all symptoms explicitly to false
        symptomConfig.forEach(({ value }) => {
          saveSymptoms[value] = false;
        });
      }
      const priorTest = isFirstTest
        ? {
            firstTest: true,
            priorTestDate: null,
            priorTestType: null,
            priorTestResult: null,
          }
        : {
            firstTest: false,
            priorTestDate: priorTestDate,
            priorTestType: priorTestType,
            priorTestResult: priorTestResult ? priorTestResult : null,
          };

      saveCallback({
        noSymptoms,
        symptoms: JSON.stringify(saveSymptoms),
        symptomOnset: onsetDate,
        ...priorTest,
        pregnancy: pregnancyResponse,
      });
      onClose();
    } else {
      e.preventDefault();
    }
  };

  const buttonGroup = (
    <div className="sr-time-of-test-buttons">
      {isModal && (
        <Button variant="unstyled" label="Cancel" onClick={onClose} />
      )}
      <Button label={saveButtonText} type={"submit"}/>
    </div>
  );

  return (
    <>
      <form onSubmit={(e) => { saveAnswers(e); }}>
        {isModal && (
          <>
            {buttonGroup}
            <h1 className="patient-name">
              {displayFullName(
                patient.firstName,
                patient.middleName,
                patient.lastName
              )}
            </h1>
          </>
        )}
        <FormGroup title="Symptoms">
          <SymptomInputs
            noSymptoms={noSymptoms}
            setNoSymptoms={setNoSymptoms}
            currentSymptoms={currentSymptoms}
            setSymptoms={setSymptoms}
            symptomListConfig={symptomConfig}
            setOnsetDate={setOnsetDate}
            onsetDate={onsetDate}
            symptomError={symptomError}
            symptomOnsetError={symptomOnsetError}
          />
        </FormGroup>

        <FormGroup title="Test History">
          <PriorTestInputs
            testTypeConfig={testConfig}
            priorTestDate={priorTestDate}
            setPriorTestDate={setPriorTestDate}
            isFirstTest={isFirstTest}
            setIsFirstTest={setIsFirstTest}
            priorTestType={priorTestType}
            setPriorTestType={setPriorTestType}
            priorTestResult={priorTestResult}
            setPriorTestResult={setPriorTestResult}
            mostRecentTest={mostRecentTest}
          />
        </FormGroup>

        {patient.gender !== "male" && (
          <FormGroup title="Pregnancy">
            <RadioGroup
              legend="Currently pregnant?"
              name="pregnancy"
              type="radio"
              onChange={(evt) => setPregnancyResponse(evt.currentTarget.value)}
              buttons={pregnancyResponses}
              selectedRadio={pregnancyResponse}
            />
          </FormGroup>
        )}
        <div className="sr-time-of-test-footer">{buttonGroup}</div>
      </form>
    </>
  );
};

AoEForm.propTypes = {};

export default AoEForm;
