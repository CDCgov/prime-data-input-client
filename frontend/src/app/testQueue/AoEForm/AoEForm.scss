div.usa-form-group--error {
  .usa-form-group--error {
    border-left-width: 0;
    padding-left: 0;

    @media screen and (min-width: 1024px) {
      border-left-width: 0.25rem;
      padding-left: 1rem;
    }
  }
}

.prime-checkboxes {
  .usa-error-message {
    + .usa-form-group--error {
      margin-top: 0;
    }
  }
}
