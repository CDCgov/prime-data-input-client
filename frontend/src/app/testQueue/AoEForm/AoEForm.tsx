import React from "react";
import { useState } from "react";
import {
  getSymptomList,
  getTestTypes,
  getPregnancyResponses,
} from "../../../patientApp/timeOfTest/constants";
import RadioGroup from "../../commonComponents/RadioGroup";
import Button from "../../commonComponents/Button";
import FormGroup from "../../commonComponents/FormGroup";
import { testResultQuery } from "../../testResults/TestResultsList";
import { useQuery } from "@apollo/client";
import moment from "moment";
import "./AoEForm.scss";
import SymptomInputs from "./SymptomInputs";
import PriorTestInputs from "./PriorTestInputs";
import { Redirect } from "react-router";

// Get the value associate with a button label
// TODO: move to utility?
const findValueForLabel = (
  label: string,
  list: { label: string; value: string }[]
) => (list.filter((item) => item.label === label)[0] || {}).value;

interface Props {
  saveButtonText: string;
  onClose?: () => void;
  patient: {
    internalId: string;
    gender: string;
  };
  facilityId: string | null;
  loadState?: {
    noSymptoms: boolean;
    symptoms: string;
    symptomOnset: string;
    priorTestDate: string;
    priorTestResult: string;
    priorTestType: string;
    firstTest: boolean;
    pregnancy: string;
  };
  saveCallback: (response: {
    noSymptoms: boolean;
    symptoms: string;
    symptomOnset: string | undefined;
    priorTestDate: string | undefined | null;
    priorTestResult: string | undefined | null;
    priorTestType: string | undefined | null;
    firstTest: boolean;
    pregnancy: string | undefined;
  }) => void;
  isModal: boolean;
  noValidation: boolean;
}

const AoEForm: React.FC<Props> = ({
  saveButtonText,
  onClose,
  patient,
  facilityId,
  loadState = {},
  saveCallback,
  isModal,
  noValidation,
}) => {
  // this seems like it will do a bunch of wasted work on re-renders and non-renders,
  // but it's all small-ball stuff for now
  const testConfig = getTestTypes();
  const symptomConfig = getSymptomList();
  const initialSymptoms: { [key: string]: boolean } = {};
  if (loadState.symptoms) {
    const loadedSymptoms: { [key: string]: string | boolean } = JSON.parse(
      loadState.symptoms
    );

    symptomConfig.forEach((opt) => {
      const val = opt.value;
      if (typeof loadedSymptoms[val] === "string") {
        initialSymptoms[val] = loadedSymptoms[val] === "true";
      } else {
        initialSymptoms[val] = loadedSymptoms[val] as boolean;
      }
    });
  } else {
    symptomConfig.forEach((opt) => {
      initialSymptoms[opt.value] = false;
    });
  }

  const [noSymptoms, setNoSymptoms] = useState(loadState.noSymptoms || false);
  const [currentSymptoms, setSymptoms] = useState(initialSymptoms);
  const [onsetDate, setOnsetDate] = useState(loadState.symptomOnset);
  const [isFirstTest, setIsFirstTest] = useState(loadState.firstTest);
  const [priorTestDate, setPriorTestDate] = useState(loadState.priorTestDate);
  const [priorTestType, setPriorTestType] = useState(loadState.priorTestType);
  const [priorTestResult, setPriorTestResult] = useState(
    loadState.priorTestResult
  );
  const [pregnancyResponse, setPregnancyResponse] = useState(
    loadState.pregnancy
  );
  const [nextPage, setNextPage] = useState(false);

  // form validation
  const [symptomError, setSymptomError] = useState<string | undefined>();
  const [symptomOnsetError, setSymptomOnsetError] = useState<
    string | undefined
  >();

  const isValidForm = () => {
    if (noValidation) return true;
    const hasSymptoms = Object.keys(currentSymptoms).some(
      (key) => currentSymptoms[key]
    );
    if (!noSymptoms && !hasSymptoms) {
      setSymptomError("Select your symptoms");
      setSymptomOnsetError(undefined);
      return false;
    }

    if (noSymptoms) {
      setSymptomError(undefined);
      setSymptomOnsetError(undefined);
      return true;
    }

    if (hasSymptoms && !onsetDate) {
      setSymptomError(undefined);
      setSymptomOnsetError("Enter the date of symptom onset");
      return false;
    }

    if (hasSymptoms && onsetDate) {
      setSymptomError(undefined);
      setSymptomOnsetError(undefined);
      return true;
    }
  };

  // TODO: only get most recent test from the backend
  const { data, loading, error } = useQuery(testResultQuery, {
    fetchPolicy: "no-cache",
    variables: { facilityId },
  });
  if (loading) return null;
  if (error) throw error;
  const mostRecentFirst = (a: any, b: any) => {
    const ma = moment(a.dateTested);
    const mb = moment(b.dateTested);
    if (ma && ma.isAfter(mb)) return -1;
    if (ma && ma.isBefore(mb)) return 1;
    return 0;
  };
  const mostRecentTest = data.testResults
    .filter((r: any) => r.patient.internalId === patient.internalId)
    .sort(mostRecentFirst)[0];

  // Auto-answer pregnancy question for males
  const pregnancyResponses = getPregnancyResponses();
  if (patient.gender === "male" && !pregnancyResponse) {
    setPregnancyResponse(findValueForLabel("No", pregnancyResponses));
  }

  const saveAnswers = (e: React.FormEvent) => {
    if (isValidForm()) {
      const saveSymptoms = { ...currentSymptoms };
      if (noSymptoms) {
        // set all symptoms explicitly to false
        symptomConfig.forEach(({ value }) => {
          saveSymptoms[value] = false;
        });
      }
      const priorTest = isFirstTest
        ? {
            firstTest: true,
            priorTestDate: null,
            priorTestType: null,
            priorTestResult: null,
          }
        : {
            firstTest: false,
            priorTestDate: priorTestDate,
            priorTestType: priorTestType,
            priorTestResult: priorTestResult ? priorTestResult : null,
          };

      saveCallback({
        noSymptoms,
        symptoms: JSON.stringify(saveSymptoms),
        symptomOnset: onsetDate,
        ...priorTest,
        pregnancy: pregnancyResponse,
      });
      if (isModal && onClose) {
        onClose();
      } else {
        setNextPage(true);
      }
    } else {
      e.preventDefault();
    }
  };

  const buttonGroup = (
    <div className="sr-time-of-test-buttons">
      <Button
        id="aoe-form-save-button"
        label={saveButtonText}
        type={"submit"}
      />
    </div>
  );

  if (nextPage) {
    return <Redirect to={"/success"} />;
  }

  return (
    <>
      <form onSubmit={saveAnswers}>
        <FormGroup title="Symptoms">
          <SymptomInputs
            noSymptoms={noSymptoms}
            setNoSymptoms={setNoSymptoms}
            currentSymptoms={currentSymptoms}
            setSymptoms={setSymptoms}
            symptomListConfig={symptomConfig}
            setOnsetDate={setOnsetDate}
            onsetDate={onsetDate}
            symptomError={symptomError}
            symptomOnsetError={symptomOnsetError}
          />
        </FormGroup>

        <FormGroup title="Test History">
          <div className="prime-formgroup__wrapper">
            <PriorTestInputs
              testTypeConfig={testConfig}
              priorTestDate={priorTestDate}
              setPriorTestDate={setPriorTestDate}
              isFirstTest={isFirstTest}
              setIsFirstTest={setIsFirstTest}
              priorTestType={priorTestType}
              setPriorTestType={setPriorTestType}
              priorTestResult={priorTestResult}
              setPriorTestResult={setPriorTestResult}
              mostRecentTest={mostRecentTest}
            />
          </div>
        </FormGroup>

        {patient.gender !== "male" && (
          <FormGroup title="Pregnancy">
            <RadioGroup
              legend="Currently pregnant?"
              name="pregnancy"
              type="radio"
              onChange={(evt) => setPregnancyResponse(evt.currentTarget.value)}
              buttons={pregnancyResponses}
              selectedRadio={pregnancyResponse}
            />
          </FormGroup>
        )}
        <div className="margin-top-4 padding-top-205 border-top border-base-lighter margin-x-neg-205">
          {buttonGroup}
        </div>
      </form>
    </>
  );
};

export default AoEForm;
