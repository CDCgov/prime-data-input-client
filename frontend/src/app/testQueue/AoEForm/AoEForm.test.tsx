import renderer, { act } from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";

import AoEForm, { LAST_TEST_QUERY } from "./AoEForm";

const mocks = [
  {
    request: {
      query: LAST_TEST_QUERY,
      variables: {
        patientId: "123",
      },
    },
    result: {
      data: {
        patient: {
          lastTest: {
            dateTested: "2021-02-05T22:01:55.386Z",
            result: "NEGATIVE",
          },
        },
      },
    },
  },
];

describe("AoEForm", () => {
  it("snapshot", async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <AoEForm
          saveButtonText="save"
          onClose={jest.fn()}
          patient={{
            internalId: "123",
            gender: "",
          }}
          saveCallback={jest.fn()}
          isModal={false}
          noValidation={true}
        />
      </MockedProvider>
    );
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
