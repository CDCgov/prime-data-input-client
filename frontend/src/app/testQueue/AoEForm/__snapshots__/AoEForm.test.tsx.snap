// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AoEForm snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="prime-formgroup"
  >
    <h2
      className="prime-formgroup-heading"
    >
      Symptoms
    </h2>
    <div
      className="usa-fieldset"
    >
      <fieldset
        className="usa-fieldset prime-checkboxes"
      >
        <legend
          className="usa-legend"
        >
          Are you experiencing any of the following symptoms?
          <span
            className="usa-sr-only"
          >
            Required
          </span>
          <span
            className="usa-hint--required"
          >
             *
          </span>
        </legend>
        <div
          className="usa-form-group"
        >
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-1"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="no"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-1"
            >
              No Symptoms
            </label>
          </div>
        </div>
      </fieldset>
      <div
        className="border-top-1px border-base-lighter margin-y-2"
      />
      <fieldset
        className="usa-fieldset prime-checkboxes symptom-checkboxes"
      >
        <legend
          className="usa-legend usa-sr-only"
        >
          What are your symptoms?
          <span
            className="usa-hint"
          >
             (optional)
          </span>
        </legend>
        <div
          className="usa-form-group"
        >
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-1"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="426000000"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-1"
            >
              Fever over 100.4F
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-2"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="103001002"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-2"
            >
              Feeling feverish
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-3"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="43724002"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-3"
            >
              Chills
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-4"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="49727002"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-4"
            >
              Cough
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-5"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="267036007"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-5"
            >
              Shortness of breath
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-6"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="230145002"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-6"
            >
              Difficulty breathing
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-7"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="84229001"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-7"
            >
              Fatigue
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-8"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="68962001"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-8"
            >
              Muscle or body aches
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-9"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="25064002"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-9"
            >
              Headache
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-10"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="36955009"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-10"
            >
              New loss of taste
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-11"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="44169009"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-11"
            >
              New loss of smell
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-12"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="162397003"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-12"
            >
              Sore throat
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-13"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="68235000"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-13"
            >
              Nasal congestion
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-14"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="64531003"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-14"
            >
              Runny nose
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-15"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="422587007"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-15"
            >
              Nausea
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-16"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="422400008"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-16"
            >
              Vomiting
            </label>
          </div>
          <div
            className="usa-checkbox"
          >
            <input
              checked={false}
              className="usa-checkbox__input"
              id="check-test-17"
              name="symptom_list"
              onChange={[Function]}
              type="checkbox"
              value="62315008"
            />
            <label
              className="usa-checkbox__label"
              htmlFor="check-test-17"
            >
              Diarrhea
            </label>
          </div>
        </div>
      </fieldset>
    </div>
    <div
      className="usa-form-group"
    >
      <label
        className="usa-label"
        htmlFor="text-test-1"
      >
        Date of symptom onset
        <span
          className="usa-hint"
        >
           (optional)
        </span>
      </label>
      <input
        aria-required="false"
        className="usa-input"
        id="text-test-1"
        max="2021-02-06"
        min="2020-02-01"
        name="symptom_onset"
        onChange={[Function]}
        type="date"
        value=""
      />
    </div>
  </div>
  <div
    className="prime-formgroup"
  >
    <h2
      className="prime-formgroup-heading"
    >
      Test History
    </h2>
    <div
      className="prime-formgroup__wrapper"
    >
      <div
        className="margin-top-2"
      >
        <b>
          Was this your most recent COVID-19 test?
        </b>
        <span
          className="usa-hint"
        >
           (optional)
        </span>
      </div>
      <p
        className="prime-previous-test-display margin-top-2 margin-bottom-0 line-height-sans-5"
      >
        <b>
          Date: 
        </b>
        Friday, February 5, 2021 10:01 PM
        <br />
        <b>
          Type: 
        </b>
        Antigen
        <br />
        <b>
          Result: 
        </b>
        NEGATIVE
      </p>
      <fieldset
        className="usa-fieldset prime-radios"
      >
        <legend
          className="usa-legend usa-sr-only"
        >
          Was this your most recent COVID-19 test?
          <span
            className="usa-hint"
          >
             (optional)
          </span>
        </legend>
        <div
          className="usa-form-group prime-radio--horizontal"
        >
          <div
            className="usa-radio prime-radio--horizontal__container"
          >
            <input
              checked={false}
              className="usa-radio__input"
              disabled={false}
              id="radio-test-1_yes_0"
              name="most_recent_flag"
              onChange={[Function]}
              type="radio"
              value="yes"
            />
            <label
              className="usa-radio__label"
              htmlFor="radio-test-1_yes_0"
            >
              Yes
            </label>
          </div>
          <div
            className="usa-radio prime-radio--horizontal__container"
          >
            <input
              checked={false}
              className="usa-radio__input"
              disabled={false}
              id="radio-test-1_no_1"
              name="most_recent_flag"
              onChange={[Function]}
              type="radio"
              value="no"
            />
            <label
              className="usa-radio__label"
              htmlFor="radio-test-1_no_1"
            >
              No
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
  <div
    className="prime-formgroup"
  >
    <h2
      className="prime-formgroup-heading"
    >
      Pregnancy
    </h2>
    <fieldset
      className="usa-fieldset prime-radios"
    >
      <legend
        className="usa-legend"
      >
        Currently pregnant?
        <span
          className="usa-hint"
        >
           (optional)
        </span>
      </legend>
      <div
        className="usa-form-group"
      >
        <div
          className="usa-radio"
        >
          <input
            checked={false}
            className="usa-radio__input"
            disabled={false}
            id="radio-test-1_77386006_0"
            name="pregnancy"
            onChange={[Function]}
            type="radio"
            value="77386006"
          />
          <label
            className="usa-radio__label"
            htmlFor="radio-test-1_77386006_0"
          >
            Yes
          </label>
        </div>
        <div
          className="usa-radio"
        >
          <input
            checked={false}
            className="usa-radio__input"
            disabled={false}
            id="radio-test-1_60001007_1"
            name="pregnancy"
            onChange={[Function]}
            type="radio"
            value="60001007"
          />
          <label
            className="usa-radio__label"
            htmlFor="radio-test-1_60001007_1"
          >
            No
          </label>
        </div>
        <div
          className="usa-radio"
        >
          <input
            checked={false}
            className="usa-radio__input"
            disabled={false}
            id="radio-test-1_261665006_2"
            name="pregnancy"
            onChange={[Function]}
            type="radio"
            value="261665006"
          />
          <label
            className="usa-radio__label"
            htmlFor="radio-test-1_261665006_2"
          >
            Would not state
          </label>
        </div>
      </div>
    </fieldset>
  </div>
  <div
    className="margin-top-4 padding-top-205 border-top border-base-lighter margin-x-neg-205"
  >
    <div
      className="sr-time-of-test-buttons"
    >
      <button
        className="usa-button"
        id="aoe-form-save-button"
        type="submit"
      >
        save
      </button>
    </div>
  </div>
</form>
`;
